chain-watch/proto/watch.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package iohk.chain.watch;
+
+
+message Block {
+  bytes content = 1;
+}
+
+message BlockSubscriptionRequest {}
+
+message TipSubscriptionRequest {}
+
+message MempoolSubscriptionRequest {}
+
+message SyncMultiverseRequest {
+  // send blocks with this chain_length or greater
+  uint32 from = 1;
+}
+
+message BlockId {
+  bytes content = 1;
+}
+
+message MempoolEvent {
+    bytes fragment_id = 1;
+    oneof event {
+        MempoolFragmentInserted inserted = 2;
+        MempoolFragmentRejected rejected = 3;
+        MempoolFragmentInABlock in_a_block = 4;
+    };
+}
+
+
+message MempoolFragmentInserted {}
+
+message MempoolFragmentRejected {
+    string reason = 1;
+}
+
+message MempoolFragmentInABlock {
+    BlockId block = 1;
+}
+
+
+service SubscriptionService {
+  // get a stream of blocks succesfully processed by the node, this means they
+  // are already validated.
+  // the parent of a block will always be streamed before the block itself.
+  rpc BlockSubscription(BlockSubscriptionRequest) returns (stream Block);
+
+  // get tip updates
+  rpc TipSubscription(TipSubscriptionRequest) returns (stream BlockId);
+
+  rpc MempoolSubscription(MempoolSubscriptionRequest) returns (stream MempoolEvent);
+
+  // fetch all blocks from the given initial chainlength to the tip, from all
+  // possible branches and in increasing order 
+  rpc SyncMultiverse(SyncMultiverseRequest) returns (stream Block);
```

Comment:
There might be consistency issues, e.g. branches starting from different roots. But I guess we just let the client to deal with that.

Reply:
I'm curious about when that can happen, you mean branches with different genesis?

Reply:
No, I meant something like that:

```
o - o - o - o - o - o - o
         \ - o - o - o
                 ^
                 |
                 We make request from this height and
                 get two incomplete branches instead
                 of a "proper" tree
```

Reply:
Until the explorer has a database I guess you need to start from 0. But after that, I was thinking on starting from `min(stability_depth, last_block_I_have)`. It may lead to sending a lot of redundant blocks, though... And you would need a way of knowing the current length (or the current stable block)

Also, I was thinking on sending all blocks of a given height before the next one, separate branches can be applied potentially in parallel, so I think it's better than sending one branch at a time.

I did write this because it was the simplest thing I could think of, though... Originally I thought about the explorer sending the list of tips it has, but that's not that simple I think.

Reply:
I've been also been wondering if this endpoint shouldn't be merged with the *live* block subscription, because otherwise there is a chance of missing a block in the middle if you switch from one stream to the other.

In that case the block subscription could have an optional parameter to sync before, and then start sending new blocks.

But clients can work around that by starting both I guess, I'm not really sure what's idiomatic here.